app/components/App/index.jsx

```diff
@@ -1,43 +1,93 @@
 import React from "react"
 import { connect } from "react-redux"
 import { I18nextProvider } from 'react-i18next'
+import Joyride from 'react-joyride'
 import { Helmet } from 'react-helmet'
 
 import i18n from '../../i18n/i18n'
 import { FlashMessages } from "../Utils"
 import { fetchCurrentUser } from '../../state/users/current_user/effects'
+import { stepSeen } from '../../state/onboarding_steps/effects'
+import { uncompletedOnboardingSteps } from '../../state/onboarding_steps/selectors'
 import { default as Sidebar } from "./Sidebar"
 import { MainModalContainer } from "../Modal/MainModalContainer"
 import PublicAchievementUnlocker from '../Users/PublicAchievementUnlocker'
 
+@connect(state => ({
+  locale: state.UserPreferences.locale,
+  onboardingSteps: uncompletedOnboardingSteps(state)
+}), {
+    fetchCurrentUser: fetchCurrentUser,
+    stepSeen: stepSeen
+  })
+export default class App extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      joyrideSteps: []
+    }
+    this.joyrideCallback = this.joyrideCallback.bind(this)
+  }
 
-@connect(state => ({locale: state.UserPreferences.locale}), {fetchCurrentUser: fetchCurrentUser})
-export default class App extends React.PureComponent {
   componentDidMount() {
     this.props.fetchCurrentUser()
   }
 
+  componentDidUpdate(prevProps, prevState) {
```

Comment:
We should use [getDerivedStateFromProps](https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops) here. It may not save us the call to `forceUpdate()` but it will at least save one render (as it is called before `render` while `componentDidUpdate` is called after)